File: Chat/ChatParser/src/SeenPacketDTO.cpp

#include "SeenPacketDTO.h"

namespace imsdk {
SeenPacketDTO::SeenPacketDTO () {}

SeenPacketDTO::SeenPacketDTO(std::string &packetId, long long messageDate, int messageStatus) {
    this->packetID = packetId;
    this->messageDate = messageDate;
    this->messageStatus = messageStatus;
}


SeenPacketDTO::~SeenPacketDTO () { }

/**
* Set the value of packetID
* @param new_var the new value of packetID
*/
void SeenPacketDTO::setPacketID(std::string new_var) {
   packetID = new_var;
}

/**
* Set the value of messageDate
* @param new_var the new value of messageDate
*/
void SeenPacketDTO::setMessageDate(long long new_var) {
   messageDate = new_var;
}

/**
* Get the value of packetID
* @return the value of packetID
*/
std::string SeenPacketDTO::getPacketID()const {
   return packetID;
}

/**
* Get the value of messageDate
* @return the value of messageDate
*/
long long SeenPacketDTO::getMessageDate()const {
   return messageDate;
}

void SeenPacketDTO::setMessageStatus(int messageStatus) {
    this->messageStatus = messageStatus;
}

int SeenPacketDTO::getMessageStatus() const {
    return messageStatus;
}

}

File: Chat/Chat/wrapper/java/JChatSDK/src/com/ringid/imsdk/SeenPacketDTO.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.ringid.imsdk;

/**
 *
 * @author root
 */
public class SeenPacketDTO {
    
    private String packetId;
    private long messageDate;
    private int messageStatus;
   
    public SeenPacketDTO(){
        
    }
    
    public SeenPacketDTO(String packetId, long messageDate, int messageStatus) {
        this.packetId = packetId;
        this.messageDate = messageDate;
        this.messageStatus = messageStatus;
    }

    /**
     * @return the packetId
     */
    public String getPacketId() {
        return packetId;
    }

    /**
     * @param packetId the packetId to set
     */
    public void setPacketId(String packetId) {
        this.packetId = packetId;
    }

    /**
     * @return the messageDate
     */
    public long getMessageDate() {
        return messageDate;
    }

    /**
     * @param messageDate the messageDate to set
     */
    public void setMessageDate(long messageDate) {
        this.messageDate = messageDate;
    }

    /**
     * @return the messageStatus
     */
    public int getMessageStatus() {
        return messageStatus;
    }

    /**
     * @param messageStatus the messageStatus to set
     */
    public void setMessageStatus(int messageStatus) {
        this.messageStatus = messageStatus;
    }
}
